### PR TITLE
Code Review 2

### PR DESCRIPTION
**Pull request by Charlie3169**
Project Description

Movie Manager allows movie connoisseurs to keep track of movies they watched and stay up to date on what they will watch next. 

* Search for movies
* Add a movie to list
* View movie list in digestible fashion
* Add specific details about movies such as genre. 
* Modify and delete entries as required.
* Mark movie as favorite for easy access
* Add scores to your watched movies and share it with your friends!

What I learned from this code review:
-Learned about Kotlin class header conventions
-Learned more about Firebase
-Learned more about Retrofit
https://kotlinlang.org/docs/coding-conventions.html

Three specific technical suggestions:
-Add JavaDoc describing what is each method and class are doing.
-Consider using RecyclerView
-Remove unused variables if you don't have plans for them

Changlelog
Updated MovieUnitTest to relect new parameters
Added UserUnitTest
Added code to UserService based off MovieService
Miscellanous cleanup

### CODE REVIEW

**Comment by Charlie3169** [comment]
Analysis of the program
Existing code was well structured and the UI looked good as well. Names were descriptive and easily understandable.

Was the program available in UC Github on time
Yes it was.

Is the program documented/commented well enough for you to understand?
Only a few comments so some of the new stuff took time to figure out.

Does the program compile? 
Yes it does.

Rationale behind your changes.
I improved the coverage of the unit tests and added to the implementation of the UserService class extrapolating from MovieService. I also cleaned up random stuff


Three Technical Concepts I learned
-Learned more about Firebase Firestore and using it.
-Learned more about using ViewHolders.
-Learned more about using Retrofit to read JSON

File: app/src/main/java/com/uc/mymovies/AuthActivity.kt [hunk 1]
```diff
@@ -11,8 +11,8 @@ import com.google.firebase.auth.FirebaseUser
 class AuthActivity : AppCompatActivity() {
 
     private var user : FirebaseUser? = null
-    private final val AUTH_REQUEST_CODE = 1337 // we are nerds
-    private final val PROVIDERS = arrayListOf(
+    private val AUTH_REQUEST_CODE = 1337 // we are nerds
```
Comment (Charlie3169) [inline app/src/main/java/com/uc/mymovies/AuthActivity.kt:14]: Apparently final was redundant

File: app/src/main/java/com/uc/mymovies/dao/UserDAO.kt [hunk 2]
```diff
@@ -1,5 +1,8 @@
 package com.uc.mymovies.dao
 
+import androidx.room.Dao
+
+@Dao
```
Comment (discospiff) [inline app/src/main/java/com/uc/mymovies/dao/UserDAO.kt:5]: Is this used in Room?  Verify that first.

If so, and if  you add @Dao, do you want to add the annotations above the methods themselves?

File: app/src/main/java/com/uc/mymovies/service/MovieService.kt [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
-import com.google.firebase.FirebaseApp
 import com.google.firebase.firestore.FirebaseFirestore
 import com.google.firebase.firestore.FirebaseFirestoreSettings
 import com.uc.mymovies.dto.Movie
 
 class MovieService {
 
-//    var firebase = FirebaseInstance.firebaseInstance
-    private lateinit var firestore : FirebaseFirestore
+    private var firestore : FirebaseFirestore
```
Comment (discospiff) [inline app/src/main/java/com/uc/mymovies/service/MovieService.kt:10]: why remove lateinit?

File: app/src/main/java/com/uc/mymovies/service/UserService.kt [hunk 4]
```diff
@@ -1,4 +1,26 @@
...
 class UserService {
+    private var firestore: FirebaseFirestore
+
+    init {
+        firestore = FirebaseFirestore.getInstance()
+        firestore.firestoreSettings = FirebaseFirestoreSettings.Builder().build()
+    }
+
+    fun fetchUsers(username: String): MutableLiveData<ArrayList<User>> {
+        //This line will probably have to be modified, I can't see your guys' firestore so I don't know your structure for it
```
Comment (discospiff) [inline app/src/main/java/com/uc/mymovies/service/UserService.kt:17]: This looks like incomplete functionality.  Only features that meet the Definition of Done should be merged to Master.

File: app/src/test/java/com/uc/mymovies/MovieUnitTest.kt [hunk 5]
```diff
@@ -5,17 +5,42 @@ import org.junit.Test
 
 import org.junit.Assert.*
 
-/**
- * Example local unit test, which will execute on the development machine (host).
- *
- * See [testing documentation](http://d.android.com/tools/testing).
- */
```
Comment (discospiff) [inline app/src/test/java/com/uc/mymovies/MovieUnitTest.kt:12]: Why remove this?  It looks like helpful KDoc

File: app/src/test/java/com/uc/mymovies/MovieUnitTest.kt [hunk 6]
```diff
@@ -5,17 +5,42 @@ import org.junit.Test
...
+
     @Test
-    fun movieDTO_matainsGenre() {
-        var movie = Movie("The Matrix", "Action/Adventure", "Who Knows", "image.com", "A movie about people who never get hit by a bullet", false);
-        assertTrue(movie.title.equals("The Matrix"));
-        assertTrue(movie.genre.equals("Action/Adventure"));
+    fun movieDTO_MaintainsState() {
+        givenMovieDTO()
+        whenMovieDTO_Initialized()
+        thenMovie_MaintainsState()
```
Comment (discospiff) [inline app/src/test/java/com/uc/mymovies/MovieUnitTest.kt:16]: Good.  Proper BDD/GWT structure.

File: app/src/test/java/com/uc/mymovies/UserUnitTest.kt [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    private fun whenUserDTO_Initialized() {
+        user = User(123456, "TheLegend27", "P@ssw0rd", "TheLegend28","eliteHaxx0r@gmail.com")
+
+    }
+
+
+    private fun givenUserDTO(){
+        user
```
Comment (discospiff) [inline app/src/test/java/com/uc/mymovies/UserUnitTest.kt:39]: What is this doing?

**Comment by discospiff** [comment]
From this review, I'd harvest the increased unit test coverage.  Or, perhaps, expand on the concept.  
The other changes... I'm not certain how they decrease technical debt, so not necessary to merge them.

**Pull request closed without merging**